packages/eslint-plugin/tests/rules/no-unnecessary-condition.test.ts

```diff
@@ -280,6 +280,11 @@ function test(a: string | null | undefined) {
     `
 function test(a: unknown) {
   return a ?? 'default';
+}
+    `,
+    `
+function test<T>(a: T) {
```

Comment:
[Testing] The cases of `T extends 'a'` (always truthy) and `T extends 'a' | null` (still needing `??`) seem to work, but aren't explicitly covered for `??` cases. Could you add in a couple more unit tests along those lines please?

Reply:
Oh sorry, I added them!
I also added a `alwaysNullish` case 😄